File: polyglot/poker_deck.go
package main

import (
	"encoding/json"
	"math/rand"
	"os"
	"time"
)

type Card struct {
	Suit string `json:"suit"`
	Rank string `json:"rank"`
}

var suits = []string{"♠", "♥", "♦", "♣"}
var ranks = []string{"A", "K", "Q", "J", "10", "9", "8", "7", "6", "5", "4", "3", "2"}

func buildDeck() []Card {
	deck := make([]Card, 0, 52)
	for _, s := range suits {
		for _, r := range ranks {
			deck = append(deck, Card{Suit: s, Rank: r})
		}
	}
	return deck
}

func draw(deck *[]Card) Card {
	d := *deck
	idx := rand.Intn(len(d))
	c := d[idx]
	d[idx] = d[len(d)-1]
	*deck = d[:len(d)-1]
	return c
}

func main() {
	rand.Seed(time.Now().UnixNano())
	deck := buildDeck()
	hand := []Card{draw(&deck), draw(&deck)}
	_ = json.NewEncoder(os.Stdout).Encode(map[string]any{
		"hand":      hand,
		"remaining": len(deck),
	})
}

File: polyglot/poker_deck.py
#!/usr/bin/env python3

"""
Poker deck + draw (Texas Hold'em-style) translation from:
backend/src/services/games/poker.ts
"""

import json
import random
import sys

SUITS = ["♠", "♥", "♦", "♣"]
RANKS = ["A", "K", "Q", "J", "10", "9", "8", "7", "6", "5", "4", "3", "2"]


def build_deck():
    return [{"suit": s, "rank": r} for s in SUITS for r in RANKS]


def draw(deck):
    idx = random.randrange(0, len(deck))
    return deck.pop(idx)


def main() -> int:
    deck = build_deck()
    hand = [draw(deck), draw(deck)]
    json.dump({"hand": hand, "remaining": len(deck)}, sys.stdout, separators=(",", ":"))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

